File: baekjoon/10815/10815.py
'''
숫자 카드 python 풀이
https://doctcoder.tistory.com/entry/%EC%88%AB%EC%9E%90-%EC%B9%B4%EB%93%9C-%ED%92%80%EC%9D%B4
'''
def bisearch(arr,target,start,end):
	while start<=end:
		mid=(start+end)//2
		if arr[mid]==target:
			return mid
		elif arr[mid]>target:
			end=mid-1
		else: start=mid+1
	return False

n=int(input())
card=list(map(int,input().split()))
m=int(input())
num=list(map(int,input().split()))
card.sort()
for i in range(m):
	if bisearch(card,num[i],0,n-1) is not False: print(1,end=' ')
	else: print(0,end=' ')

File: baekjoon/10815/10815.cpp
/*
숫자 카드 C++ 풀이
https://doctcoder.tistory.com/entry/%EC%88%AB%EC%9E%90-%EC%B9%B4%EB%93%9C-%ED%92%80%EC%9D%B4
*/
#include <iostream>
#include <vector>
#include <algorithm>
using namespace std;
int main()
{
  int n,m,a;
	cin>>n;
	vector<int>nv;
	vector<int>mv;
	for(int i=0;i<n;i++)
	{
		cin>>a;
		nv.push_back(a);
	}
	cin>>m;
	for(int i=0;i<m;i++)
	{
		cin>>a;
		mv.push_back(a);
	}
	sort(nv.begin(),nv.end());
	for(int i=0;i<m;i++)
	{
		if (binary_search(nv.begin(),nv.end(),mv[i])) cout<<1<<" ";
		else cout<<0<<" ";
	}
}
